frameworks/nodejs/src/main/java/io/dekorate/nodejs/NodeInfoReader.java

```diff
@@ -0,0 +1,175 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.dekorate.nodejs;
+
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.concurrent.atomic.AtomicInteger;
+import java.util.regex.Pattern;
+
+import io.dekorate.DekorateException;
+import io.dekorate.Logger;
+import io.dekorate.LoggerFactory;
+import io.dekorate.project.BuildInfo;
+import io.dekorate.project.BuildInfoBuilder;
+import io.dekorate.project.BuildInfoReader;
+import io.dekorate.utils.Exec;
+import io.dekorate.utils.Exec.ProjectExec;
+import io.dekorate.utils.Git;
+import io.dekorate.utils.Strings;
+
+public class NodeInfoReader implements BuildInfoReader {
+
+  private final Logger LOGGER = LoggerFactory.getLogger();
+
+  private static final String NPM = "npm";
+
+  private static final String PACKAGE_JSON = "package.json";
+
+  private static final String SRC = "src";
+
+  private static final String OPEN_BRACKET = "{";
+  private static final String CLOSE_BRACKET = "}";
+  private static final String QUOTE = "'";
+
+  private static final String EQUALS = "=";
+  private static final String COLON = ":";
+  private static final String DASH = "-";
+  private static final String DOT = ".";
+
+  private static final String NAME = "name";
+  private static final String VERSION = "version";
+  private static final String BIN = "bin";
+  private static final String MAIN = "main";
+
+  public static final String NEW_LINE = "[\\n\\r]+";
+
+  @Override
+  public int order() {
+    return 500;
+  }
+
+  @Override
+  public boolean isApplicable(Path root) {
+    return root.resolve(PACKAGE_JSON).toFile().exists();
+  }
+
+  @Override
+  public BuildInfo getInfo(Path root) {
+    Path packageJson = root.resolve(PACKAGE_JSON);
+    Map<String, String> properties = new HashMap<>();
+    properties.putAll(readPackageJson(packageJson));
+
+    String name = properties.getOrDefault(NAME, properties.getOrDefault(NAME, root.getFileName().toString()));
+    String version = properties.get(VERSION);
+    String main = properties.get(MAIN);
+
+    StringBuilder sb = new StringBuilder();
+    sb.append(name);
+    if (Strings.isNotNullOrEmpty(version)) {
+      sb.append(DASH).append(version);
+    }
+
+    if (version == null) {
+      LOGGER.warning("Could not detect project version. Using 'latest'.");
+      version = "latest";
+    }
+
+
+    return new BuildInfoBuilder()
+      .withName(name)
+      .withVersion(version)
+      .withPackaging("js")
+      .withBuildTool(NPM)
+      .withBuildToolVersion(getVersion(root))
+      .withOutputFile(root.resolve("app.js"))
+      .withClassOutputDir(null)
+      .withResourceDir(null)
+      .build();
+  }
+
+  /**
+   * Parse build.gradle and read the jar configuration as a {@link Map}.
```

Comment:
```suggestion
   * Parse package.json and read the jar configuration as a {@link Map}.
```
Also it would be nice to have a comment on what exactly is parses